### DISCUSSION TITLE
Unable to shutdown or reboot when using step ssh proxycommand

### DISCUSSION THREAD

**Discussion by jodygilbert** [smallstep/certificates#1143] [General]
I'm using step-cli 0.22.0 on Windows 10 to connect to various Ubuntu servers.  The servers are running Ubuntu 18.04 and 20.04.

I've noticed that if I reboot or shutdown the servers the connection hangs, it appears to be indefinitely but I haven't left it for more than a couple of minutes.

If I do this from a Ubuntu client the reboot works fine.

To get around this I've added the following options to the ssh config file, which forces the client to timeout and disconnect:

	ServerAliveInterval 15
	ServerAliveCountMax 5

I cannot tell if the issue is with the step client or the ssh client, any suggestions greatly appreciated.

**Comment by maraino** [comment]
Hi @jodygilbert, I'm able to reproduce this, I will investigate.

**Reply by maraino** [reply]
The problem seems to be that the workgroup replicating the STDIN does not finish, but the STDOUT does. And when you exit with `ctrl-d`, both finish. I will move this to an issue in [smallstep/cli](https://github.com/smallstep/cli).

**Reply by maraino** [reply]
In any case, if I press a key, or perhaps `Enter`, the connection closes.

**Reply by maraino** [reply]
https://github.com/smallstep/cli/issues/788

**Comment by jodygilbert** [comment]
Thanks for the quick response and the suggested workaround, I can stop searching google for fixes in the ssh client.
